views/default/navigation/menu/user_hover/contents.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+// g = guid, pog = page_owner_guid, c = contexts, m = mac
+$guid = (int) get_input("g", 0, false);
+$page_owner_guid = (int) get_input("pog", 0, false);
+$contexts = (array) get_input("c", array(), false);
+$mac = get_input("m", "", false);
+$input = (array) get_input("i", array(), false);
+
+// verify MAC
+$data = serialize(array($guid, $page_owner_guid, $contexts, $input));
+if ($mac !== hash_hmac("sha256", $data, get_site_secret())) {
+	return;
+}
+
+$user = get_user($guid);
+if (!$user) {
+	return;
+}
+
+if ($page_owner_guid) {
+	// set correct page_owner
+	elgg_set_page_owner_guid($page_owner_guid);
+}
+
+// set correct contexts
```

Comment:
currently we do not have a good way to do that... should i pop until get_context is empty?